### DISCUSSION TITLE
yaml vs smart_thermostat services

### DISCUSSION THREAD

**Discussion by arkadi222** [ScratMan/HASmartThermostat#88] [Q&A]
Hi
it's quite strange, that some parameters need to be changed by yaml, some of them has set_xxx service :)

1. could you explain (or complete tutorial) which parameters is set by functions, which by yaml  (for example: set_pid_gain has no ke parameter?, set_preset_temp has no required parameter )
2. how to remove some preset_mode? (if I don't want to use home_temp any more, is there any way to drop from HA database by set_preset_temp because even if I comment home_temp in yaml I see this preset in custom:simple-thermostat card)
3.  there are some attributes set by yaml, but I don't see them in dev tools/states (for example min_cycle_duration or sampling_period). How can I check this values was changed? (in yaml, after restart)
4. I see some entity attributes (pid_p,pid_i,pid_d,pid_e) and didn't find it in documentation - what do they mean?

**Comment by ovizii** [comment]
I am also wondering why I see pid_i when checking the state with the developer tools even though I never enabled debugging nor have I set this value.

**Reply by ScratMan** [reply]
The pid_i is always visible because we need to store it in the database to restore it upon restarting HA.

**Comment by ScratMan** [comment]
> Hi it's quite strange, that some parameters need to be changed by yaml, some of them has set_xxx service :)
> 
> 1. could you explain (or complete tutorial) which parameters is set by functions, which by yaml  (for example: set_pid_gain has no ke parameter?, set_preset_temp has no required parameter )

The "hardware related" parameters (cycle duration, temperature limits, sensors and heaters entities...) are only set by YAML as they shouldn't have to be changed during operation. Presets are adjustable by actions (services) because you may want to adjust your setpoints sometimes, and doing it through actions doesn't need to restart.
It the same for PID gains adjustment, they are available (even Ke) through action to make the tuning easier without requiring to restart HA.


> 2. how to remove some preset_mode? (if I don't want to use home_temp any more, is there any way to drop from HA database by set_preset_temp because even if I comment home_temp in yaml I see this preset in custom:simple-thermostat card)

Just use the `xxx_temp_disable: true` in set preset temp service to disable preset names xxx.

```YAML
action: smart_thermostat.set_preset_temp
data:
  eco_temp_disable: true
target:
  entity_id: climate.salle_de_bain
```


> 3. there are some attributes set by yaml, but I don't see them in dev tools/states (for example min_cycle_duration or sampling_period). How can I check this values was changed? (in yaml, after restart)

They are not available inside UI. They are loaded at start and can't be changed in UI, so the value you set is the one used.

> 4. I see some entity attributes (pid_p,pid_i,pid_d,pid_e) and didn't find it in documentation - what do they mean?

It's the actual PID values computed by the thermostat. It's there because the `debug: true` argument is set in YAML, and helps to adjust the gains, but it may overload your database. The debug option should be removed once the thermostat is set correctly and the PID gains should be backup in the YAML to save values in case of a database crash.
